scripts/setup/nrfconnect/update_ncs.py

```diff
@@ -0,0 +1,126 @@
+
+def print_messages(messages : list, yellow_text :  bool):
+    # Add colour formatting if yellow text was set
+    if yellow_text:
+        messages = [f"\33[33m{message}\x1b[0m" for message in messages]
+
+    for message in messages:
+        print(message)
+
+def print_check_version_warning_message(current_version, recommended_version):
+    current_version_message = f"WARNING: Your current NCS version ({current_version})"
+    recommended_version_message = f"differs from the recommended ({recommended_version})."
+    allowed_message = "Please be aware that it may lead to encountering unexpected problems."
+    update_message = "Consider updating NCS to the recommended version, by calling:"
+    call_command_message = os.path.abspath(__file__) + " --update"
+
+    # Get the longest message lenght, to fit warning frame size.
+    longest_message_len = max([len(current_version_message), len(recommended_version_message), len(allowed_message), len(update_message), len(call_command_message)])
+
+    # To keep right frame shape the space characters are added to messages shorter than the longest one.
+    fmt = "# {:<%s}#" % (longest_message_len)
+
+    print_messages([(longest_message_len+3)*'#', fmt.format(current_version_message), fmt.format(recommended_version_message), fmt.format(''),
+                    fmt.format(allowed_message), fmt.format(update_message), fmt.format(call_command_message), (longest_message_len+3)*'#'], sys.stdout.isatty())
+
+def main():
+
+    try:
+        zephyr_base = os.getenv("ZEPHYR_BASE")
+        if not zephyr_base:
+            raise RuntimeError("No ZEPHYR_BASE environment variable found, please set ZEPHYR_BASE to a zephyr repository path.")
+
+        parser = argparse.ArgumentParser(description='Script helping to update nRF Connect SDK to currently recommended version.')
+        parser.add_argument("-c", "--check", help="Check if your current nRF Connect SDK version is the same as recommended one.", action="store_true")
+        parser.add_argument("-u", "--update", help="Update your nRF Connect SDK to currently recommended version.", action="store_true")
+        args = parser.parse_args()
+
+        ncs_base = os.path.join(zephyr_base, '../nrf')
+        recommended_version = get_ncs_recommended_version()
+
+        if args.check:
+            print("Checking current nRF Connect SDK version...")
+
+            current_version = get_repository_commit_sha(ncs_base)
+
+            if current_version == recommended_version:
+                print("Your current version is up to date with the recommended one.")
+            else:
+                print_check_version_warning_message(current_version, recommended_version)
+                sys.exit(1)
+
+        if args.update:
+            print("Updating nRF Connect SDK to recommended version...")
+            update_ncs(ncs_base, recommended_version)
+
+    except (RuntimeError, subprocess.CalledProcessError) as e:
+        print(e)
```

Comment:
if there is just handling the error msg then you should add `sys.exit(1)` to be sure that something goes wrong